File: java/lib/src/main/java/algorithm/sorting/BubbleSort.java
package algorithm.sorting;

import java.util.Collections;
import java.util.List;

/**
 * BubbleSort
 */
public class BubbleSort extends Sort {
  @Override
  protected List<Integer> sort(List<Integer> arr, SortType type) {
    int length = arr.size();

    for (int i = 0; i < length; i++) {
      for (int j = 0; j < length - i - 1; j++) {
        if (super.compare(arr.get(j), arr.get(j + 1), type)) {
          Collections.swap(arr, j, j + 1);
        }
      }
    }

    return arr;
  }
}

File: kotlin/lib/src/main/kotlin/algorithm/sorting/BubbleSort.kt
package algorithm.sorting

public class BubbleSort : Sort() {
  override fun sort(arr: Array<Int>, type: SortType): Array<Int> {
    val length = arr.size

    for (i in 0..length - 1) {
      for (j in i + 1..length - 1) {
        if (compare(arr[i], arr[j], type)) {
          arr.swap(i, j)
        }
      }
    }

    return arr
  }
}
